qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1371,6 +1371,36 @@ def substitute_node(self, node, op, inplace=False):
             self._decrement_op(node.op)
         return new_node
 
+    def swap_nodes(self, node1, node2):
+        """Swap connected nodes e.g. due to commutation.
+
+        Args:
+            node1 (OpNode): predecessor node
+            node2 (OpNode): successor node
+
+        Raises:
+            DAGCircuitError: if either node is not an OpNode
+        """
+        if not (isinstance(node1, DAGOpNode) and isinstance(node2, DAGOpNode)):
+            raise DAGCircuitError("nodes to swap are not both OpNodes")
```

Comment:
```suggestion
            raise DAGCircuitError("nodes to swap are not both DAGOpNodes")
```
(so the user can more easily find the correct type, if they see this error)